test/models/test_models.py

```diff
@@ -137,23 +140,19 @@ def _train(model):
         self.assertEqual(model.encoder.state_dict(), encoder_current_state_dict)
         self.assertNotEqual(model.head.state_dict(), head_current_state_dict)
 
-    @pytest.fixture(autouse=True)
-    def inject_fixtures(self, caplog):
-        self._caplog = caplog
-
-    def test_roberta_bundler_get_model(self):
+    @patch('logging.Logger.warning')
+    def test_roberta_bundler_get_model(self, mock):
         from torchtext.models import RobertaEncoderConf, RobertaBundle
-
-        with self._caplog.at_level(logging.WARNING):
-            dummy_encoder_conf = RobertaEncoderConf(
+        dummy_encoder_conf = RobertaEncoderConf(
                 vocab_size=10, embedding_dim=16, ffn_dimension=64, num_attention_heads=2, num_encoder_layers=2
             )
-            model_bundle = RobertaBundle(dummy_encoder_conf)
-            model_bundle.get_model(
-                load_weights = False,
-                freeze_encoder = True
-            )
-            self.assertTrue(self._caplog.records[0].message.startswith('The encoder is not loaded'))
+        model_bundle = RobertaBundle(dummy_encoder_conf)
+        model_bundle.get_model(
+            load_weights = False,
+            freeze_encoder = True
+        )
+        mock.assert_called_with(ANY)
```

Comment:
Are you sure you do not want to check the message argument?

Reply:
Added message for stricter test.